crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/CompareContributionService.java

```diff
@@ -41,19 +37,19 @@ public boolean shouldCreateContribution(CalculateContributionDTO calculateContri
 
         if (activeContribution.isPresent()
                 && areContributionRecordsIdentical(contributionResult, activeContribution.get())
-                && !(hasMagsCourtOutcomeChanged(magsCourtOutcome, repOrderDTO)
-                    || CaseType.APPEAL_CC.getCaseTypeString().equals(repOrderDTO.getCatyCaseType()))) {
+                && isMagsCourtOutcomeUnchanged(calculateContributionDTO.getMagCourtOutcome(),
+                                               calculateContributionDTO.getRepOrderDTO())) {
             log.info("Contributions should not be created");
             return false;
         }
         log.info("Contributions should be created");
         return true;
     }
 
-    private boolean hasMagsCourtOutcomeChanged(MagCourtOutcome magsCourtOutcome, RepOrderDTO repOrderDTO) {
-        if (repOrderDTO == null) return false;
+    private boolean isMagsCourtOutcomeUnchanged(MagCourtOutcome magsCourtOutcome, RepOrderDTO repOrderDTO) {
+        if (repOrderDTO == null) return true;
 
-        return !Objects.equals(magsCourtOutcome, MagCourtOutcome.getFrom(repOrderDTO.getMagsOutcome()));
+        return Objects.equals(magsCourtOutcome, MagCourtOutcome.getFrom(repOrderDTO.getMagsOutcome()));
```

Comment:
Similar question to before on the object equality check here.

Reply:
See comment above.